components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/NatsMessageConsumer.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.inbound.endpoint.protocol.nats;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Polling consumer for NATS to initialize connection and poll messages.
+ */
+public class NatsMessageConsumer {
+
+    private static final Log log = LogFactory.getLog(NatsMessageConsumer.class.getName());
+    private NatsInjectHandler injectHandler;
+    private Properties natsProperties;
+    private NatsMessageListener natsMessageListener;
+    private String subject;
+    private String injectingSequenceName;
+
+    public NatsMessageConsumer(Properties natsProperties, String injectingSequenceName) {
+        this.natsProperties = natsProperties;
+        this.injectingSequenceName = injectingSequenceName;
```

Comment:
the name passed for the 'injectingSequenceName' parameter is the 'inboundendpointName' at 'NatsEndpointManager.startEndpoint()'. I feel like there is an issue there.